github3/repos/branch.py

```diff
@@ -28,5 +29,51 @@ def _update_attributes(self, branch):
         #: Provides the branch's protection status.
         self.protection = branch.get('protection')
 
+        if 'self' in self.links:
+            self._api = self.links['self']
+        else:  # Branches obtained via `repo.branches` don't have links.
+            base = self.commit.url.split('/commit', 1)[0]
+            self._api = self._build_url('branches', self.name, base_url=base)
+
     def _repr(self):
         return '<Repository Branch [{0}]>'.format(self.name)
+
+    def protect(self, enforcement=None, status_checks=None):
+        """Enable force push protection and configure status check enforcement.
+
+        See: http://git.io/v4Gvu
+
+        :param str enforcement: (optional), Specifies the enforcement level of
+            the status checks. Must be one of 'off', 'non_admins', or
+            'everyone'. Use `None` or omit to use the already associated value.
+        :param list status_checks: (optional), An list of strings naming
+            status checks that must pass before merging. Use `None` or omit to
+            use the already associated value.
+        """
+        previous_values = self.protection['required_status_checks']
+        if enforcement is None:
+            enforcement = previous_values['enforcement_level']
+        if status_checks is None:
+            status_checks = previous_values['contexts']
+
+        edit = {'protection': {'enabled': True, 'required_status_checks': {
+            'enforcement_level': enforcement, 'contexts': status_checks}}}
+        json = self._json(self._patch(self._api, data=dumps(edit),
+                                      headers=self.PREVIEW_HEADERS), 200)
+
+        # When attempting to clear `contexts`, the reply from github doesn't
+        # currently reflect the actual value. Let's fix that for now.
+        cur_contexts = self.protection['required_status_checks']['contexts']
+        if status_checks == [] != cur_contexts:
```

Comment:
It may not look like it but in python it is correct. Try `0 == 0 != True`.
I'm checking that were attempting to clear the value and the value is
different than the previously set value.